File: 41.First_missing_positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:

        #list comprehension to retrieve positive numbers from the list
        nums = [n for n in nums if n > 0]

        #sorting the list 
        nums.sort()
        
        #the target is always 1 cuz it the lowest possible positive number
        target = 1

        for n in nums:
            if n == target:
                target += 1
            elif n > target:
                return target
        return target

File: 41.first_missing_positive.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var firstMissingPositive = function(nums) {
    nums.sort((a,b) => a -b);

    let target = 1;
    for (let i = 0; i < nums.length; i++){
        if (nums[i] > 0 && nums[i] === target){
            target ++;
        }
        else if(nums[i] > target){
            return target;
        }
    }
    return target
};